File: source_code/app/routers/recommendations.js
const router = require('express').Router();
const { getRecommendations } = require('../services/recommendationService');
const { getCurrentUser } = require('../services/authService');

router.get('/', async (req, res) => {
  try {
    const user = await getCurrentUser(req);
    const limit = Math.min(parseInt(req.query.limit) || 6, 20);
    res.json(await getRecommendations(user._id, limit));
  } catch (e) {
    res.status(e.status || 500).json({ detail: e.message });
  }
});

module.exports = router;

File: source_code/app/routers/recommendations.py
from fastapi import APIRouter, Depends, Query

from app.services.recommendation_service import get_recommendations
from app.services.auth_service import get_current_user

router = APIRouter()


@router.get("/", summary="Get personalized product recommendations")
async def recommendations(
    limit: int = Query(6, ge=1, le=20),
    current_user: dict = Depends(get_current_user),
):
    return await get_recommendations(str(current_user["_id"]), limit=limit)
